bootstrap/cli/ci/updater.py

```
@@ -170,8 +170,25 @@ async def run():
+        if ci.raw_artifacts is not None:
+            public_url = ci.raw_artifacts.public_url
+        if public_url is None and ci.storage is not None:
+            public_url = ci.storage.public_url
+        if public_url:
+            url = f"{public_url}/build-dependencies/py27.zip"
         output.parent.mkdir(parents=True, exist_ok=True)
-        urllib.request.urlretrieve(url, output)
+        request = urllib.request.Request(
+            url,
+            headers={"User-Agent": "Mozilla/5.0 (compatible; EFA-CI/1.0)"},
+        )
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)